src/lightning_app/core/app.py

```diff
@@ -520,6 +526,84 @@ def state_dict(self) -> Dict:
     def load_state_dict(self, state: Dict) -> None:
         self.set_state(state)
 
+    @classmethod
+    def _load_checkpoint_from_json_file(cls, path: str) -> None:
+        checkpoint_dict = json.load(open(path))
+        # TODO: Checkpoint versioning and compatibility checks
+        return checkpoint_dict
+
+    def _get_checkpoint_if_available_locally(self, checkpoint: str) -> Optional[dict]:
+        checkpoint_dict = None
+        if os.path.exists(checkpoint):
+            checkpoint_dict = self._load_checkpoint_from_json_file(checkpoint)
+        if os.path.exists(os.path.join(self.checkpoint_dir, checkpoint)):
+            checkpoint_dict = self._load_checkpoint_from_json_file(os.path.join(self.checkpoint_dir, checkpoint))
+
+        return checkpoint_dict
+
+    def _get_checkpoint_if_available_on_drive(self, checkpoint: str) -> Optional[dict]:
+        os.makedirs(self.checkpoint_dir, exist_ok=True)
+        drive = Drive("lit://checkpoints", component_name="root", root_folder=self.checkpoint_dir)
+
+        with _context(ComponentContext.WORK):
+            found_checkpoints = drive.list()
+            logger.debug("Found checkpoints: %s", found_checkpoints)
+
+        checkpoint_drive_path = ""
+
+        if checkpoint == "latest":
+            checkpoint_filenames_regex = r"lightningapp_checkpoint_([0-9]{10}).json"
+            r = re.compile(checkpoint_filenames_regex)
+            filtered_list = list(filter(r.search, found_checkpoints))
+            checkpoint_drive_path = max(filtered_list, key=lambda i: int(r.search(i).group(1))) if filtered_list else ""
+        else:
+            for cp in found_checkpoints:
+                if checkpoint in cp:
+                    checkpoint_drive_path = cp
+                    break
```

Comment:
Can we make this a function? This would make it much easier to unit-test the cases.

Reply:
added in an utility function